cirq-google/cirq_google/engine/virtual_engine_factory.py

```diff
@@ -367,3 +370,36 @@ def create_noiseless_virtual_engine_from_latest_templates() -> SimulatedLocalEng
     processor_ids = list(MOST_RECENT_TEMPLATES.keys())
     template_names = [MOST_RECENT_TEMPLATES[k] for k in processor_ids]
     return create_noiseless_virtual_engine_from_templates(processor_ids, template_names)
+
+
+def create_noisy_virtual_engine_from_processor_id_and_simulator_class(
+    processor_id: str, simulator_class: Type[SimulatesSamples], **kwargs
+) -> SimulatedLocalEngine:
+    """Creates a virtual engine with a noisy simulator based on a processor id.
+
+    Args:
+        processor_id: The string name of a processor that has available noise data.
+        simulator_class: The class of the type of simulator to be initialized.
```

Comment:
Also note that this simulator should support the noise parameter, since this is not strictly guaranteed by the SimulatesSamples interface.